multicluster/cmd/gateways.go

```diff
@@ -77,8 +77,21 @@ func newGatewaysCommand() *cobra.Command {
 				os.Exit(1)
 			}
 
+			leases, err := k8sAPI.CoordinationV1().Leases(multiclusterNs.Name).List(cmd.Context(), metav1.ListOptions{})
```

Comment:
is the a label we can use to get all the `service-mirror-write-*` lease resources?  just so that we don't have to assume that this namespace will never contain leases for any other purpose.

Reply:
I have included labels precisely for this reason:
https://github.com/linkerd/linkerd2/blob/b7e0be6e671a06d6711668571ff323fd4a8fd8fb/multicluster/cmd/service-mirror/main.go#L183-L190

It's a bit annoying but `client-go` doesn't respect the metadata on the lease lock. When a lease is first acquired, if it doesn't exist it gets created (see https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L329-L343).

The [Create](https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/resourcelock/leaselock.go#L56-L66) implementation on the Lease lock re-creates the metadata and passes only the namespace and name. So unfortunately, the metadata we put on the Lease doesn't serve any purpose.

I thought we had two options here:
* Do nothing for now
* When a Lease is acquired, check if it has the labels, and if it doesn't, update the Lease. If we do it when the Lease is acquired the write is protected.

Since we don't have any informers and indexers for Lease objects, I thought the simplest option is to do nothing and avoid the API Server overhead from doing a read (and possibly a write). I thought it would be unlikely for now to have other leases in the namespace. A label selector is the right approach here, I was just unsure if it's worth doing now.

Reply:
Ah, that's unfortunate.  As a compromise, we could fetch all leases like we do here, but then ignore any of them which don't have a name of the form "service-mirror-write-*".